arroyo/processing/strategies/dead_letter_queue/dead_letter_queue.py

```diff
@@ -33,17 +33,17 @@ def __init__(
     def poll(self) -> None:
         try:
             self.__next_step.poll()
-        except InvalidMessage as e:
-            self.__metrics.increment("dlq.received_message")
-            self.__policy.handle_invalid_message(e)
+        except InvalidMessages as e:
+            self.__metrics.increment("dlq.received_message", len(e.messages))
+            self.__policy.handle_invalid_messages(e)
 
     def submit(self, message: Message[TPayload]) -> None:
         assert not self.__closed
         try:
             self.__next_step.submit(message)
-        except InvalidMessage as e:
-            self.__metrics.increment("dlq.received_message")
-            self.__policy.handle_invalid_message(e)
+        except InvalidMessages as e:
+            self.__metrics.increment("dlq.received_message", len(e.messages))
+            self.__policy.handle_invalid_messages(e)
```

Comment:
Here you can see the DLQ only accepts one message at a time, and then submits that message to the next step. It is just prepared to handle an exception with multiple broken messages if need be.